File: primality test.py
from math import isqrt

def is_prime(n: int) -> bool:
    if n <= 3:
        return n > 1
    if n % 2 == 0 or n % 3 == 0:
        return False
    limit = isqrt(n)
    for i in range(5, limit+1, 6):
        if n % i == 0 or n % (i+2) == 0:
            return False
    return True

print("This is a program to see if the given integer, be it large or small is a prime number.")
print()
n = eval(input("Enter the integer: "))

if is_prime(n):
    print("The Given integer is prime number")
else:
    print("The given integer contains more than 2 factors")

print("Number -> ",n)

File: primality test.c
/**
 * Copyright (c) 2024 Adwaith
 * 
 * This software is released under the MIT License.
 * https://opensource.org/licenses/MIT
 */

#include <stdio.h>

int input;

int IsPrime(int n)
{
    if (n == 2 || n == 3)
        return 0;

    if (n <= 1 || n % 2 == 0 || n % 3 == 0)
        return 1;

    for (int i = 5; i * i <= n; i += 6)
    {
        if (n % i == 0 || n % (i + 2) == 0)
            return 1;
    }

    return 0;
}

int main() {
    printf("Type a number: ");
    scanf("%d", &input);
    if (IsPrime(input)) {
        printf("\nThe Given integer is prime number\n");
    } else {
        printf("\n The Given integer has more than 2 factors");
    }
    printf("Number -> ");
    printf("%d", input);
}